README.md

```diff
@@ -150,6 +152,16 @@ const env = cleanEnv(process.env, {
 });
 ```
 
+### TypeScript users
+
+You can use either one of `makeBaseValidator`, `makeExactValidator` and `makeMarkupValidator`
+depending on your use case:
+
+- `makeBaseValidator<BaseT>` when you want the output to be narrowed-down to a subtype of `BaseT` (e.g. `str`).
+- `makeExactValidator<T>` when you want the output to be widened to `T` (e.g. `bool`).
+- `makeMarkupValidator` for input which can produce arbitrary output types (e.g. `json`).
```

Comment:
I don't know if it makes sense to expose `makeMarkupValidator` in the public API. What other use cases can you think of for it? If we keep it private for now we can always expose it later. Given the domain of "parsing env var strings" I think json() would be the dominant case

Reply:
I was thinking query parameters, e.g. `"option1=true&option2=false&token=XXX"`; or POSIX like flags e.g. `-b -c --long-flag=/bin/xxx`, `/etc/fstab`, XML, YAML... JSON5; honestly I don't know much about backend developers practices , but I had imagined exposing this one would give plenty of flexibility for consumers of this library, freeing them from the hassle of typings these validators correctly. 

**PS1**: didn't test but I think we should also forbid specs with choices in markup validators, correct? 
**PS2**: I went back and forth between naming this validator `markupValidator` and `structuredValidator` ; English is not my native language and if you have better naming ideas I'm all for it

Reply:
I think `structuredValidator` would be a better name for sure, when most people hear "markup" they probably think of a specific type of format like HTML/XML/etc. 

Also re `choices`, a nice side effect of removing `structuredValidator` from the public API is we don't have to worry about that interaction :)

Reply:
> Also re choices, a nice side effect of removing structuredValidator from the public API is we don't have to worry about that interaction :)

Well, with one important exception: `json` validator. In any case, I did implement this specific logic here: 
328e05642a486264a6031bf381527102df72ceef

Yet, I'm happy to _not_ export `makeStructuredValidator` if you don't want to.